modules/tide_edit_protection/tide_edit_protection.module

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains tide_edit_protection.module.
+ */
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\views\ViewExecutable;
+
+/**
+ * Implements hook_help().
+ */
+function tide_edit_protection_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    case 'help.page.tide_edit_protection':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Protects unsaved change.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function tide_edit_protection_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $include = FALSE;
+  $config = \Drupal::config('tide_edit_protection.form');
+
+  // Check entity form conditions.
+  $entity_method = $config->get('entity_method');
+  if (!empty($entity_method)) {
+    $formObject = $form_state->getFormObject();
+    if (method_exists($formObject, 'getEntity')) {
+      $entity = $formObject->getEntity();
+      $entity_type = $entity->getEntityTypeId();
+      $entity_bundle = $entity_type . '__' . $entity->bundle();
+      if ('all' === $entity_method) {
+        $include = TRUE;
+      }
+      elseif ('entity_type' === $entity_method) {
+        $entity_types = $config->get('entity_types');
+        if (in_array($entity_type, $entity_types)) {
+          $include = TRUE;
+        }
+      }
+      elseif ('entity_bundle' === $entity_method) {
+        $entity_bundles = $config->get('entity_bundles');
+        if (in_array($entity_bundle, $entity_bundles)) {
+          $include = TRUE;
+        }
+      }
+    }
+  }
+
+  // Check other forms by form_id, support Regex.
+  if (FALSE === $include) {
```

Comment:
Has there been a team decision to use Yoda conditionals? I think SDP's "unofficial" (a discussion in Slack in a channel far, far away) code style preference is not to use them.

Reply:
Thanks @GROwen . oh~ sorry, forgot about this . removed.

Reply:
To confirm, no Yoda conditionals. Thanks for changing.